pkg/eraser/eraser.go

```diff
@@ -195,11 +195,16 @@ func removeImages(c Client, targetImages []string) error {
 		}
 	}
 
-	// map for non-running imags by name
-	nonRunningNames := make(map[string]struct{}, len(allImages)-len(runningImages))
-	for key := range nonRunningImages {
-		if idMap[key] != nil && len(idMap[key]) > 0 {
-			nonRunningNames[idMap[key][0]] = struct{}{}
+	log.V(1).Info("List of non-running images by digest", "nonRunningImages", nonRunningImages)
+	log.V(1).Info("Map of digest to image name(s)", "idMap", idMap)
+
+	// mapping of non-running image names to digests
+	nonRunningNames := make(map[string]string, len(allImages)-len(runningImages))
```

Comment:
Can you add a unit test for what's being fixed here?

Reply:
Will do, thanks

- [ ] Unit test

Reply:
Separated into another PR, which will be put up when the test is done, so marking as resolved.